CompletableFuture doesn't raise exceptions for runtime errors

**Description**

Errors happening after FFMpeg's startup don't raise exceptions.
FFMpeg startup errors are raised correctly.

**Steps to reproduce**

1. Run an Nginx server with nginx-rtmp-module installed (such as the `alfg/nginx-rtmp` docker image)
2. Run an FFMpeg process with the following command `ffmpeg -f lavfi -i testsrc=size=1920x1080:rate=25 -f lavfi -i sine=f=1:b=2000 -shortest -f flv rtmp://localhost:1935/proxy/source`
3. Run the following test class

```java
import com.github.kokorin.jaffree.StreamType;
import com.github.kokorin.jaffree.ffmpeg.FFmpeg;
import com.github.kokorin.jaffree.ffmpeg.UrlInput;
import com.github.kokorin.jaffree.ffmpeg.UrlOutput;

class TestClass {
    public static void main(String[] args) {
        FFmpeg.atPath()
              .addInput(UrlInput.fromUrl("rtmp://localhost:1935/hls/source"))
              .addOutput(UrlOutput.toUrl("rtmp://localhost:1935/hls/target")
                                  .setFormat("flv")
                                  .addMap(0, StreamType.AUDIO)
                                  .addMap(0, StreamType.VIDEO)
                                  .setCodec(StreamType.AUDIO, "copy")
                                  .setCodec(StreamType.VIDEO, "copy")
              )
              .executeAsync()
              .toCompletableFuture()
              .exceptionally(error -> {
                  System.err.println("An error has occured");
                  error.printStackTrace();

                  return null;
              })
              .join();
    }
}
```

4. Stop the Nginx server

This error message will come up but no exception will be raised.

> c.g.k.jaffree.ffmpeg.FFmpegResultReader  : [error] rtmp://localhost:1935/hls/source: Input/output error

**Expected behavior**

An exception should be raised because this issue causes FFMpeg to be stopped.

**Logs**

> c.g.k.jaffree.ffmpeg.FFmpegResultReader  : [error] rtmp://localhost:1935/hls/source: Input/output error

I've found an easy solution ! 🥳

I've stumbled upon this FFMpeg argument while working on how to reproduce the error in Jaffree's tests :

> -xerror (global)
Stop and exit on error

This works perfectly 👍

```java
import com.github.kokorin.jaffree.StreamType;
import com.github.kokorin.jaffree.ffmpeg.FFmpeg;
import com.github.kokorin.jaffree.ffmpeg.UrlInput;
import com.github.kokorin.jaffree.ffmpeg.UrlOutput;

class TestClass {

    public static void main(String[] args) {
        FFmpeg.atPath()
              .addArgument("-xerror")
              .addInput(UrlInput.fromUrl("rtmp://localhost:1935/proxy/source"))
              .addOutput(UrlOutput.toUrl("rtmp://localhost:1935/proxy/target")
                                  .setFormat("flv")
                                  .addMap(0, StreamType.AUDIO)
                                  .addMap(0, StreamType.VIDEO)
                                  .setCodec(StreamType.AUDIO, "copy")
                                  .setCodec(StreamType.VIDEO, "copy")
              )
              .execute();
    }
}
```

Which produces these logs.

> 2022-01-28 17:23:31.537  INFO   --- [         StdErr] c.g.k.jaffree.ffmpeg.FFmpegResultReader  : [info] frame=  133 fps= 88 q=-1.0 size=    1052kB time=00:00:05.35 bitrate=1609.8kbits/s speed=3.53x    
2022-01-28 17:23:32.038  INFO   --- [         StdErr] c.g.k.jaffree.ffmpeg.FFmpegResultReader  : [info] frame=  158 fps= 78 q=-1.0 size=    1236kB time=00:00:06.31 bitrate=1602.2kbits/s speed=3.12x    
2022-01-28 17:23:32.546  INFO   --- [         StdErr] c.g.k.jaffree.ffmpeg.FFmpegResultReader  : [info] frame=  186 fps= 74 q=-1.0 size=    1430kB time=00:00:07.43 bitrate=1574.8kbits/s speed=2.95x    
2022-01-28 17:23:33.051  INFO   --- [         StdErr] c.g.k.jaffree.ffmpeg.FFmpegResultReader  : [info] frame=  224 fps= 74 q=-1.0 size=    1708kB time=00:00:08.95 bitrate=1561.7kbits/s speed=2.95x    
2022-01-28 17:23:33.278  INFO   --- [         StdErr] c.g.k.jaffree.ffmpeg.FFmpegResultReader  : [info] frame=  261 fps= 74 q=-1.0 size=    1977kB time=00:00:10.43 bitrate=1551.1kbits/s speed=2.95x    
2022-01-28 17:23:33.279 ERROR   --- [         StdErr] c.g.k.jaffree.ffmpeg.FFmpegResultReader  : [error] rtmp://localhost:1935/proxy/source: Input/output error
2022-01-28 17:23:33.282  INFO   --- [         StdErr] c.g.k.jaffree.ffmpeg.FFmpegResultReader  : [info] Conversion failed!
2022-01-28 17:23:33.283  INFO   --- [           main] c.g.k.jaffree.process.ProcessHandler     : Process has finished with status: 1
2022-01-28 17:23:33.283  WARN   --- [         StdErr] c.g.kokorin.jaffree.process.Executor     : Interrupting starter thread (main) because of exception: ffmpeg exited with message: [error] rtmp://localhost:1935/proxy/source: Input/output error
2022-01-28 17:23:33.284  WARN   --- [           main] c.g.k.jaffree.process.ProcessHandler     : Process has been interrupted
2022-01-28 17:23:33.284  INFO   --- [           main] c.g.k.jaffree.ffmpeg.FFmpegStopper       : Ignoring class java.io.IOException: Stream closed
Exception in thread "main" com.github.kokorin.jaffree.JaffreeException: Failed to execute, was interrupted
	at com.github.kokorin.jaffree.process.ProcessHandler.interactWithProcess(ProcessHandler.java:189)
	at com.github.kokorin.jaffree.process.ProcessHandler.execute(ProcessHandler.java:146)
	at com.github.kokorin.jaffree.ffmpeg.FFmpeg.execute(FFmpeg.java:400)
	at Scratch.main(scratch_1.java:19)
	Suppressed: com.github.kokorin.jaffree.JaffreeException: ffmpeg exited with message: [error] rtmp://localhost:1935/proxy/source: Input/output error
		at com.github.kokorin.jaffree.ffmpeg.FFmpegResultReader.read(FFmpegResultReader.java:136)
		at com.github.kokorin.jaffree.ffmpeg.FFmpegResultReader.read(FFmpegResultReader.java:38)
		at com.github.kokorin.jaffree.process.ProcessHandler$1.run(ProcessHandler.java:233)
		at com.github.kokorin.jaffree.process.Executor$1.run(Executor.java:83)
		at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at com.github.kokorin.jaffree.process.ProcessHandler.waitForExecutorToStop(ProcessHandler.java:318)
	at com.github.kokorin.jaffree.process.ProcessHandler.interactWithProcess(ProcessHandler.java:182)
	... 3 more
>
> Process finished with exit code 1

--- a/README.md
+++ b/README.md
@@ -41,7 +41,7 @@ Inspired by [ffmpeg-cli-wrapper](https://github.com/bramp/ffmpeg-cli-wrapper)
 
 ## Checking media streams with ffprobe
 
-See whole example [here](/src/test/java/examples/ShowStreamsExample.java).
+See whole example [here](src/test/java/examples/ShowStreamsExample.java).
 
 ```java
 FFprobeResult result = FFprobe.atPath()
@@ -60,7 +60,7 @@ for (Stream stream : result.getStreams()) {
 
 Sometimes ffprobe can't show exact duration, use ffmpeg trancoding to NULL output to get it.
 
-See whole example [here](/src/test/java/examples/ExactDurationExample.java).
+See whole example [here](src/test/java/examples/ExactDurationExample.java).
 
 ```java
 final AtomicLong durationMillis = new AtomicLong();
@@ -83,7 +83,7 @@ System.out.println("Exact duration: " + durationMillis.get() + " milliseconds");
 
 ## Re-encode and track progress
 
-See whole example [here](/src/test/java/examples/ReEncodeExample.java).
+See whole example [here](src/test/java/examples/ReEncodeExample.java).
 
 ```java
 final AtomicLong duration = new AtomicLong();
@@ -118,7 +118,7 @@ FFmpeg.atPath()
 
 Pay attention that arguments related to Input must be set at Input, not at FFmpeg.
 
-See whole example [here](/src/test/java/examples/CutAndScaleExample.java).
+See whole example [here](src/test/java/examples/CutAndScaleExample.java).
 
 ```java
 FFmpeg.atPath()
@@ -139,7 +139,7 @@ FFmpeg.atPath()
 
 ## Custom parsing of ffmpeg output
 
-See whole example [here](/src/test/java/examples/ParsingOutputExample.java).
+See whole example [here](src/test/java/examples/ParsingOutputExample.java).
 
 ```java
 // StringBuffer - because it's thread safe
@@ -165,7 +165,7 @@ System.out.println("Loudnorm report:\n" + loudnormReport);
 Ability to interact with SeekableByteChannel is one of the features, which distinct Jaffree from 
 similar libraries. Under the hood Jaffree uses tiny FTP server to interact with SeekableByteChannel.
 
-See whole example [here](/src/test/java/examples/UsingChannelsExample.java).
+See whole example [here](src/test/java/examples/UsingChannelsExample.java).
 ```java
 try (SeekableByteChannel inputChannel =
          Files.newByteChannel(pathToSrc, StandardOpenOption.READ);
@@ -188,7 +188,7 @@ requires seekable output for many formats.
 
 Under the hood pipes are not OS pipes, but TCP Sockets. This allows much higher bandwidth.
 
-See whole example [here](/src/test/java/examples/UsingStreamsExample.java).
+See whole example [here](src/test/java/examples/UsingStreamsExample.java).
 
 ```java
 try (InputStream inputStream =
@@ -209,7 +209,7 @@ try (InputStream inputStream =
 
 ## Screen Capture
 
-See whole example [here](/src/test/java/examples/ScreenCaptureExample.java).
+See whole example [here](src/test/java/examples/ScreenCaptureExample.java).
 
 ```java
 FFmpeg.atPath()
@@ -239,8 +239,8 @@ Files.move(pathToOptimized, pathToVideo, StandardCopyOption.REPLACE_EXISTING);
 
 ## Produce Video in Pure Java Code
 
-See whole example [here](/src/test/java/examples/ProduceVideoExample.java). 
-Check also more [advanced example](/src/test/java/examples/BouncingBallExample.java) which produce 
+See whole example [here](src/test/java/examples/ProduceVideoExample.java). 
+Check also more [advanced example](src/test/java/examples/BouncingBallExample.java) which produce 
 both audio and video 
 
 ```java
@@ -283,11 +283,11 @@ FFmpeg.atPath()
 
 Here is an output of the above example:
 
-![example output](/src/test/resources/examples/programmatic.gif)
+![example output](src/test/resources/examples/programmatic.gif)
 
 ### Consume Video in Pure Java Code
 
-See whole example [here](/src/test/java/examples/ExtractFramesExample.java).
+See whole example [here](src/test/java/examples/ExtractFramesExample.java).
 
 ```java
 FFmpeg.atPath()
@@ -333,10 +333,28 @@ FFmpeg.atPath()
         .execute();
 ```
 
+## Managing errors
+
+Jaffree will raise exceptions when a fatal error that causes a non-zero exit code occurs.
+
+In some cases an error can occur but FFmpeg manages to catch it and exit correctly. This can be a 
+convenient case, although sometimes one would prefer an exception to be raised to Jaffree.
+
+To do so, the [`-xerror`](https://ffmpeg.org/ffmpeg.html#Advanced-options) argument can be used to
+tell FFmpeg to exit the process with an error status when an error occurs.
+
+```java
+FFmpeg.atPath()
+      .addArgument("-xerror")
+      // ...
+```
+
+Please see [Issue 276](https://github.com/kokorin/Jaffree/issues/276) for more details on an actual
+usecase.
 
 ## FFmpeg stop
 
-See whole examples [here](/src/test/java/examples/StopExample.java).
+See whole examples [here](src/test/java/examples/StopExample.java).
 
 ### Grace stop
 
@@ -398,7 +416,7 @@ thread.interrupt();
 
 ## Java 8 Completion API
 
-See whole examples [here](/src/test/java/examples/CompletionExample.java).
+See whole examples [here](src/test/java/examples/CompletionExample.java).
 
 ```java
 ffmpeg.executeAsync().toCompletableFuture()
@@ -500,7 +518,7 @@ FFmpegResult result = FFmpeg.atPath(BIN)
 ## Programmatic mosaic video creation
 
 Jaffree allows simultaneous reading from several sources (with one instance per every source and target).
-You can find details in  Mosaic [example](/src/test/java/examples/MosaicExample.java).
+You can find details in  Mosaic [example](src/test/java/examples/MosaicExample.java).
 
 # Build & configuration
 

